### ISSUE TITLE
Unexpected client exception regarding buffers

### ISSUE THREAD

**Issue by cavallium** [Telegram4J/Telegram4J#12]
```java
2022-05-07 01:05:36.603 ERROR 4175655 DefaultMTProtoClient : [C:0x71f96fdf] Unexpected client exception
java.lang.IndexOutOfBoundsException: readerIndex(4) + length(458562960) exceeds writerIndex(108): CompositeByteBuf(ridx: 4, widx: 108, cap: 108, components=2)
        at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1442)
        at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1428)
        at io.netty.buffer.AbstractByteBuf.readSlice(AbstractByteBuf.java:879)
        at telegram4j.mtproto.transport.IntermediateTransport.decode(IntermediateTransport.java:45)
        at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106)
        at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
        at reactor.core.publisher.DrainUtils.postCompleteDrain(DrainUtils.java:136)
        at reactor.core.publisher.DrainUtils.postComplete(DrainUtils.java:187)
        at reactor.core.publisher.FluxBufferPredicate$BufferPredicateSubscriber.onComplete(FluxBufferPredicate.java:356)
        at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete(FluxMap.java:269)
        at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete(FluxMap.java:269)
        at reactor.netty.channel.FluxReceive.terminateReceiver(FluxReceive.java:468)
        at reactor.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:260)
        at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:404)
        at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:419)
        at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:473)
        at reactor.netty.channel.ChannelOperations.onInboundClose(ChannelOperations.java:434)
        at reactor.netty.channel.ChannelOperationsHandler.channelInactive(ChannelOperationsHandler.java:73)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248)
        at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)
        at io.netty.channel.AbstractChannel$AbstractUnsafe$7.run(AbstractChannel.java:813)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by skykatik** [comment]
I can't reproduce it. Have you received any more such exceptions? And on what supposedly request did this appear?

**Comment by cavallium** [comment]
here is the log: https://local.cavallium.it/mirrors/latest.txt

**Comment by skykatik** [comment]
I will close the issue soon, because the error has been fixed in Telegram4J@c86ac9f.
